packages/contracts-bedrock/test/L2/CrossL2Inbox.t.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Testing utilities
+import { CommonTest } from "test/setup/CommonTest.sol";
+import { EIP1967Helper } from "test/mocks/EIP1967Helper.sol";
+import { Reverter, ConfigurableCaller } from "test/mocks/Callers.sol";
+import { ICrossL2Inbox } from "src/L2/ICrossL2Inbox.sol";
+
+// Libraries
+import { SafeCall } from "src/libraries/SafeCall.sol";
+
+import { L1Block } from "src/L2/L1Block.sol";
+
+contract CrossL2InboxTest is CommonTest {
+    address depositor;
+
+    function setUp() public virtual override {
+        super.setUp();
+        depositor = l1Block.DEPOSITOR_ACCOUNT();
+    }
+
+    function testFuzz_executeMessage_succeeds(
+        bytes calldata _msg,
+        ICrossL2Inbox.Identifier calldata _id,
+        address _target
+    )
+        external
+        payable
+    {
+        // need to make sure address leads to successfull SafeCall by executeMessage
+        // TODO: how to make sure gasLeft coincides wtih executeMessage?
+        bool success = SafeCall.call({ _target: _target, _gas: gasleft(), _value: msg.value, _calldata: _msg });
+        vm.assume(success);
+
+        // timestamp invariant
+        vm.assume(_id.timestamp <= block.timestamp);
+
+        // chainId invariant
+        uint256[] memory chainIds = new uint256[](1);
+        chainIds[0] = _id.chainId;
+        vm.prank(depositor);
+        l1Block.setL1BlockValues(0, 0, 0, bytes32(0), 0, bytes32(0), 0, 0, chainIds);
```

Comment:
I think its easier to use `vm.mockCall` rather than setting things int he actual L1 block contract